Modules/Server/Sources/PocketCastsServer/Public/Discover/DiscoverServerHandler.swift

```diff
@@ -101,55 +106,101 @@ public class DiscoverServerHandler {
         .eraseToAnyPublisher()
     }
 
-    private let tokenHelper = {
-        let connection = URLConnection(handler: URLSession.shared)
-        return TokenHelper(urlConnection: connection)
-    }()
+    /// A method to check whether the response from the source URL authenticated successfully.
+    /// - Parameters:
+    ///   - item: An item which is `authenticated == true`
+    /// - Returns: Whether or not the authentication succeeded
+    public func checkSourceAuthentication(for item: DiscoverItem) async -> Bool {
+        // If there's no source URL, consider authentication failed. This shouldn't happen.
+        guard item.authenticated == true, let source = item.source else {
```

Comment:
Nit: instead of checking against the optional `authenticated` I'm wondering if `DiscoverItem` can have a computed property `isAuthenticated` that unwraps it or fallback to false